server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivationDisabled.java

```diff
@@ -0,0 +1,139 @@
+    public static Collection<Object[]> data() {
+        String[] databaseTypes = PersistenceIT.DATABASE.getDatabaseTypes();
+        List<Object[]> params = new ArrayList<>(databaseTypes.length);
+        for (String databaseType : databaseTypes) {
+            params.add(new Object[]{databaseType});
+        }
+        return params;
+    }
+
+    public JdbcStringBasedCacheStorePassivationDisabled(String databaseType) {
+        this.database = PersistenceIT.DATABASE.getDatabase(databaseType);
+    }
+
+    @Test(timeout = 600000)
+    public void testFailoverWithPassivationDisabled() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC)
+                .createPersistenceConfiguration(database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder());
+
+        cache.put("k1", "v1");
+        cache.put("k2", "v2");
+        // test passivation==false, database should contain all entries which are in the cache
+        assertNotNull(table.getValueByKey("k1"));
+        assertNotNull(table.getValueByKey("k2"));
+
+        SERVERS.getServerDriver().stop(0);
+        SERVERS.getServerDriver().restart(0);
+
+        assertNotNull(table.getValueByKey("k1"));
+        assertNotNull(table.getValueByKey("k2"));
+        assertEquals("v1", cache.get("k1"));
+        assertEquals("v2", cache.get("k2"));
+        //when the entry is removed from the cache, it should be also removed from the cache store (the store
+        //and the cache are the same sets of keys)
+        cache.remove("k1");
+        assertNull(table.getValueByKey("k1"));
+        cache.clear();
+        table.releaseConnection();
+    }
+
+    @Test(timeout = 600000)
+    public void testPreloadWithoutPurge() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC)
+                .createPersistenceConfiguration(database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder());
+
+        cache.clear();
+        cache.put("k1", "v1");
+        cache.put("k2", "v2");
+
+        assertNotNull(table.getValueByKey("k1"));
+        assertNotNull(table.getValueByKey("k2"));
+
+        SERVERS.getServerDriver().stop(0);
+        SERVERS.getServerDriver().restart(0);
+
+        // test preload==true, entries should be immediately in the cache after restart
+        assertEquals(2, cache.size());
```

Comment:
This isn't testing that preload works, as it is just checking the cache size which will include the loaders. Instead you should assert that the entry is in-memory by doing `cache.withFlags(Flag.SKIP_CACHE_LOAD).size()`